.github/workflows/docker.yml

```diff
@@ -69,6 +70,7 @@ jobs:
           if [ "${{ github.event_name }}" = "push" ]; then
             TAGS="${DOCKER_IMAGE}:latest"
           fi
+          echo ::set-output name=gradleVersionTag::${VERSION}
```

Comment:
The release version in Gradle is now expected to be the form X.Y.Z (with an optional -SNAPSHOT suffix).  I think it would be best if you got rid of version resolution via $GITHUB_REF and use the `./gradle currentVersion` mechanism I described in my previous comment.

Reply:
I've seen approaches like this run into problems in the past: developers are free to push tags without any restrictions GitHub can implement, and it's nondeterministic which of multiple tags for the current HEAD such client-side tooling will select. When it comes to GitHub actions being triggered by the `release` events though, the maintainer creating the GitHub release selected a specific tag to use/create and GitHub is providing that to the workflow as a singular input via `GITHUB_REF`. The real world case here is developers pushing pre-release tags for testing, and then the GitHub release flow arbitrarily picking one of those to tag a release with rather than what the maintainer creating the official release tag selected.

Besides, this was already how the current workflow was sourcing the Docker image tag to apply and these changes are just also routing that same designation into the container build. So now they're both chained off the same input and it would be a broader scope of change to source that version designation differently. @bdferris-v2 I'd be happy to take that change on within this PR though if that's still preferred given the gotcha I described above.

The one substantial change I DID make here was dropping the `v` prefix when reading the Git tag. While it **is** a git convention to prefix version number tags with `v`, the convention with Docker image tags is to drop it. The `-Prelease.forceVersion=` option was also expecting it without the `v` prefix, so I did it upstream here to drive both the Docker tag and the Gradle version. So as of right now if this got merged, future docker container image publishings would drop the`v`—which seems like a good move to me this early in publishing one to bring in line with convention.

Reply:
I wouldn't describe it as non-deterministic.  The `axios-release-plugin` has a [defined strategy](https://axion-release-plugin.readthedocs.io/en/latest/configuration/version/#first-tag-encountered) for resolving the current version from the Git commit history.  It's smart enough to ignore arbitrary tags like `i-love-tacos` but it is true that it will complain if you create a version tag like `v1.tacos`.

So I'm agreed that this requires us to use consistent vX.Y.Z version tags when tagging releases, but that doesn't seem too terrible.  More importantly, this wasn't introduced arbitrarily.  The `jpackage` app packaging phase specifically fails if passed a non-X.Y.Z version number, which is what led to this upstream change in the first place.  We are going to need some consistency in our version tags one way or another.  If there are other approaches that play well with the free-wheeling nature of Git versioning, I'm interested in learning more about them.

But if we stick with what we've got, then I'd argue that you should drop the code to resolve the version from `GITHUB_REF` and instead use the `currentVersion` approach I discussed earlier so that we get consistent version numbers whether we are building from a release tag, a PR, etc.  I'll admit it's a hack (thus my inclination to just include the .git metadata in the Docker build itself, but I digress).

Unfortunately, while playing around with `forceVersion` behavior, I discovered that if you pass in `1.2.3-SNAPSHOT` as the forced version, the plugin will happily append an additional -SNAPSHOT to make the version `1.2.3-SNAPSHOT-SNAPSHOT`.  So there's probably an argument for making the version resolution command:

```
$(./gradlew cV -q -Prelease.quiet | sed 's/-SNAPSHOT//')
```